ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoTransactionBuilder.kt

```diff
@@ -0,0 +1,228 @@
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param notary The notary that will be used to notarise the specified [ContractState].
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, notary: Party): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param contractId The class name of the [Contract] associated with the transaction state.
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, contractId: String): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param contractId The class name of the [Contract] associated with the transaction state.
+     * @param notary The notary that will be used to notarise the specified [ContractState].
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, contractId: String, notary: Party): UtxoTransactionBuilder
+
+    /**
+     * Adds an output state to the current [UtxoTransactionBuilder].
+     *
+     * @param contractState The [ContractState] to add to the current [UtxoTransactionBuilder].
+     * @param contractId The class name of the [Contract] associated with the transaction state.
+     * @param notary The notary that will be used to notarise the specified [ContractState].
+     * @param encumbrance The index of an associated, encumbered state, or null if no encumbrance applies to the associated transaction state.
+     * @return Returns a [UtxoTransactionBuilder] including the additional output state.
+     */
+    fun addOutputState(contractState: ContractState, contractId: String, notary: Party, encumbrance: Int?): UtxoTransactionBuilder
+
+    /**
+     * Sets the transaction time window to be valid from the specified [Instant], tending towards positive infinity.
+     *
+     * @param from The [Instant] from which the transaction time window is valid.
+     * @return Returns a [UtxoTransactionBuilder] including the specified time window.
+     */
+    fun setTimeWindowFrom(from: Instant): UtxoTransactionBuilder
```

Comment:
I don't think the notary would be happy with that. 
In order to be able to archive (consumed) states, we always need a finite end of the time window for every tx. That can be months in the future, but it needs to be defined.

Reply:
@mnesbit TimeWindow itself on UtxoTransactionBuilder is non-nullable now, but it's components (From and Until) are nullable. What are your thoughts?